### PR TITLE
Add support for u-boot auto TFTP boot

### PR DESCRIPTION
**Pull request by alistair23**
This patch series does a few things:

1. Updates the sdcard image to include the standard /boot/ directory. This is common for OE built SD card images. At the moment though as u-boot doesn't support SD/MMC access we can't load the images from here.
2. Add support for SMP boot on the FreedomU540
3. Build u-boot by default and make it easy to set an override to include it as an OpenSBI payload.
4. Setup scripts to help with a TFTP autoboot.

### CODE REVIEW

File: recipes-bsp/u-boot/u-boot_2019.04.bbappend [hunk 1]
```diff
@@ -18,4 +18,36 @@ SRC_URI += "file://0001-riscv-add-infrastructure-for-calling-functions-on-ot.pat
...
+
+DEPENDS += "u-boot-tools-native"
+
+UBOOT_ENTRYPOINT = "0x80200000"
+
+# Overwrite this for your server
+TFTP_SERVER_IP ?= "127.0.0.1"
+
+u_boot_do_patch() {
+      sed -i -e 's,@SERVERIP@,${TFTP_SERVER_IP},g' ${WORKDIR}/0001-sifive-fu540-Set-default-arguments-to-help-booting.patch
```
Comment (kraj) [inline recipes-bsp/u-boot/u-boot_2019.04.bbappend]: Can we do it after patching ? say via do_configure_prepend, editing patches seems quite cryptic to me.

Comment (alistair23) [reply recipes-bsp/u-boot/u-boot_2019.04.bbappend]: I did think of that, but that seems just as messy. If you prefer that I'll give it a shot.

Comment (kraj) [reply recipes-bsp/u-boot/u-boot_2019.04.bbappend]: hmm how many such instances are there ?

Comment (alistair23) [reply recipes-bsp/u-boot/u-boot_2019.04.bbappend]: Just 1, it actually seems easy. Just testing it now

**Comment by alistair23** [comment]
V2 sent

**Comment by alistair23** [comment]
Is this ok to be merged?

**Pull request merged**